langchain-core/src/runnables/base.ts

```diff
@@ -2783,3 +2804,83 @@ export class RunnablePick<
     return IterableReadableStream.fromAsyncGenerator(wrappedGenerator);
   }
 }
+
+export interface RunnableToolLikeArgs<
+  RunInput extends z.ZodType = z.ZodType,
+  RunOutput = unknown
+> extends Omit<RunnableBindingArgs<z.infer<RunInput>, RunOutput>, "config"> {
+  name: string;
+
+  description?: string;
+
+  schema: RunInput;
+
+  config?: RunnableConfig;
+}
+
+export class RunnableToolLike<
```

Comment:
Should this extend `RunnableLambda` instead?

Reply:
Spoke to nuno and he suggested `RunnableBinding`, which a few other runnables extend too